vmaas/reposcan/exporter.py

```diff
@@ -748,6 +749,59 @@ def _dump_oval(self, dump):
                 dump.execute("insert into oval_module_test_detail values (?, ?)",
                              (oval_module_test_id, module_stream))
 
+    def _dump_csaf(self, dump):
+        """Select CSAF data"""
+        # Dump data from csaf_product table
+        dump.execute("""CREATE TABLE IF NOT EXISTS csaf_product (
+                        id INTEGER PRIMARY KEY AUTOINCREMENT,
+                        cpe_id INTEGER NOT NULL,
+                        package_name_id INTEGER,
+                        package_id INTEGER,
+                        module_stream_id INTEGER
+                    )""")
+        with self._named_cursor() as cursor:
+            cursor.execute("""SELECT *
+                            FROM csaf_product""")
+            for csaf_product_id, cpe_id, package_name_id, package_id, module_stream in cursor:
+                dump.execute("INSERT INTO csaf_product VALUES (?, ?, ?, ?, ?)",
+                             (csaf_product_id, cpe_id, package_name_id, package_id, module_stream))
+
+        # Dump data from csaf_cve_product table
+        dump.execute("""CREATE TABLE IF NOT EXISTS csaf_cve_product (
+                        id INTEGER PRIMARY KEY AUTOINCREMENT,
+                        cve_id INTEGER NOT NULL,
+                        csaf_product_id INTEGER NOT NULL,
+                        csaf_product_status_id INTEGER NOT NULL,
```

Comment:
It would be beneficial to have `cve` string instead of `cve_id` in the map in vmaas-lib so we don't need to translate id to name during the evaluation. I had a quick look and it seems that the current vmaas-lib code does use only cve strings. It can be done either here by replacing `cve_id INTEGER NOT NULL` with `cve TEXT NOT NULL` and joining with `cve` table before the insert, or it can be done during the load in vmaas-lib.

Reply:
I haven't read whole vmaas-lib repo, just took a look at first existing functions and it operates with cve_id then translates it to cve name in cache `CveNames  map[int]string`. Then cve details are accessed in another cache `CveDetail map[string]CveDetail`.

Reply:
It will work, there's no problem. I just though we might want to use the CVE string directly so we won't need to add another for loop for every list of cve_ids. I was just looking at how current evaluation works and it seems to be using cve names directly instead of returning ids and then converting it to cve names. We don't seem to be operating with cve_ids at all in vmaas-lib, only with cve names.